addons/mail/static/src/model/model_field_command.js

```diff
@@ -9,65 +9,196 @@ odoo.define('mail/static/src/model/model_field_command.js', function (require) {
 }
 
 /**
  * Returns a increment command to give to the model manager at create/update.
+ * This command can be used for attribute fields (number typed).
+ * The field value will be increase by `amount`.
  *
  * @param {number} [amount=1]
+ * @returns {FieldCommand}
  */
 function increment(amount = 1) {
-    return new FieldCommand((field, record, options) => {
-        const oldValue = field.get(record);
-        return field.set(record, oldValue + amount, options);
-    });
+    return new FieldCommand('increment', amount);
+}
+
+/**
+ * Return a insert command to give to the model manager at create/update.
+ * `insert` command can be used for relation fields.
+ * - Create and replace a new record for an x2one field
+ *      if `data` does not match the currect field value;
+ * - or update the record for an x2one field
+ *      if `data` matches the currect field value;
+ * - or create and insert new records for an x2many field
+ *      if `data` does not match any existing records in the field values;
+ * - or update the records for an x2many field
+ *      if `data` matches existing records in the field values.
+ *
+ * @param {Object|Object[]} data  - data object or data objects array to insert record(s).
+ * @returns {FieldCommand}
+ */
+function insert(data) {
+    return new FieldCommand('insert', data);
+}
+
+/**
+ * Return a insert-and-replace command to give to the model manager at create/update.
+ * `insertAndReplace` command can be used for relation fields.
+ * - Create and replace a new record for an x2one field
+ *      if `data` does not match the current field value;
+ * - or update the record for an x2one field
+ *      if `data` matches currect field value;
+ * - or create and insert new records for an x2many field
+ *      if `data` does not match any existing record in the field values
+ *      and then replace old records with the new ones`;
+ * - or update the record for an x2many field
+ *      if `data` matches existing records in the field values
+ *      and then replace old records with the updated ones.
+ *
+ * @param {Object|Object[]} data - data object or data objects array to insert and replace record(s).
+ * @returns {FieldCommand}
+ */
+function insertAndReplace(data) {
+    return new FieldCommand('insert-and-replace', data);
+}
+
+/**
+ * Return a link command to give to the model manager at create/update.
+ * `link` command can be used for relation fields.
+ * - Replace the current value by `newValue` for an x2one field;
+ * - or add `newValue` to its value for an x2many field.
+ *
+ * @param {mail.model|mail.model[]} newValue  - record or records array to be linked.
+ * @returns {FieldCommand}
+ */
+function link(newValue) {
+    return new FieldCommand('link', newValue);
+}
+
+/**
+ * Return a replace command to give to the model manager at create/update.
+ * `replace` command can be used for relation fields.
+ * - Replace the current value by `newValue` for an x2one/x2many field.
```

Comment:
It doesn't help to describe replace command by using the replace word in the doc.
It should say it does unlink of all the existing value that are not in new value, and then link all the value from new value that were missing.
(And then for x2many it sorts them to have them in they order they are given in new value)